src/clients/c++/concurrency_manager.cc

```diff
@@ -49,28 +49,62 @@ nic::Error
 ConcurrencyManager::Create(
     const int32_t batch_size, const size_t max_threads,
     const size_t sequence_length, const bool zero_input,
+    const std::unordered_map<std::string, std::vector<int64_t>>& input_shapes,
     const std::string& data_directory,
     const std::shared_ptr<ContextFactory>& factory,
     std::unique_ptr<LoadManager>* manager)
 {
-  manager->reset(new ConcurrencyManager(
+  std::unique_ptr<ConcurrencyManager> local_manager(new ConcurrencyManager(
       batch_size, max_threads, sequence_length, zero_input, factory));
 
+  std::unique_ptr<nic::InferContext> ctx;
+  RETURN_IF_ERROR(local_manager->factory_->CreateInferContext(&ctx));
+
+  // Validate user provided shape
+  if (!input_shapes.empty()) {
+    // Get all input names
+    for (const auto& input : ctx->Inputs()) {
+      auto it = input_shapes.find(input->Name());
+      if (it != input_shapes.end()) {
+        bool valid = true;
+
+        const auto& dims = it->second;
+        const auto& config_dims = input->Dims();
+        if (((int64_t)dims.size()) != config_dims.size()) {
+          valid = false;
+        } else {
+          for (size_t idx = 0; idx < dims.size(); idx++) {
+            if ((dims[idx] <= 0) ||
```

Comment:
Seems like it would be better to check that user-specified dims are all positive when we process the arg